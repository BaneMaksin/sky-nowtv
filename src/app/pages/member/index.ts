/*
 * Date: 7/15/22, 8:33 AM
 * Copyright: Branislav Maksin (c) 2022
 */

// eslint-disable-next-line import/no-default-export, no-restricted-exports
export { MemberRoutes as default } from './routes';
