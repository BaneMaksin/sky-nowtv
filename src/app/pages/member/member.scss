/*!
 * Date: 7/15/22, 8:33 AM
 * Copyright: Branislav Maksin (c) 2022
 */

.title {
  color: var(--content-title-color);
  margin: 0 0 14px;
}
