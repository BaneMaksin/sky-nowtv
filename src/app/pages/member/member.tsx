/*
 * Date: 7/15/22, 8:33 AM
 * Copyright: Branislav Maksin (c) 2022
 */

import React from 'react';
import { useParams } from 'react-router-dom';

import { Grid, MoreButton, useLoadMoreMessages } from '@shared';
import { MappedMessage } from '@store/messages';

import { useCheckMemberId, useStoreData } from './hooks';
import { MemberMessage } from './message';
import { MemberHeader } from './header';

// Style
import style from './member.scss';

/**
 * Member page component.
 *
 * @returns Virtual element.
 */
export function MemberPage(): JSX.Element {

  // Get member ID from route params
  const { memberId } = useParams();

  // Check if member id is in valid format
  const isMemberIdValid = useCheckMemberId(memberId);

  // Handle batch messages load
  const { chunk, memoizedClickHandler } = useLoadMoreMessages();

  // Get messages and member data
  const { messages, member, isMore } = useStoreData(chunk, memberId, isMemberIdValid);

  // Render view
  return messages?.length && member ? (
    <section>
      <h1 className={style.title}>
        Member
      </h1>

      <MemberHeader member={member} />

      <h3 className={style.title}>
        Messages
      </h3>

      <Grid>
        {messages.map((message: MappedMessage) => (
          <MemberMessage
            key={message.id}
            data={message}
          />
        ))}
      </Grid>

      <MoreButton
        clickHandler={memoizedClickHandler}
        isMore={isMore}
      />
    </section>
  ) : null;
}
