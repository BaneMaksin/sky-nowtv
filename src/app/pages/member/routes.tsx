/*
 * Date: 7/15/22, 8:33 AM
 * Copyright: Branislav Maksin (c) 2022
 */

import React from 'react';
import { Routes, Route, Navigate } from 'react-router-dom';

import { MemberPage } from './member';

/**
 * Member routes component.
 *
 * @returns Virtual element.
 */
export const MemberRoutes = (): JSX.Element => (
  <Routes>
    <Route
      element={<MemberPage />}
      path=":memberId"
    />
    <Route
      element={(
        <Navigate
          replace
          to="/"
        />
      )}
      path="*"
    />
  </Routes>
);
