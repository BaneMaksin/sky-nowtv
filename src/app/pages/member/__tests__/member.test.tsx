/*
 * Date: 7/15/22, 8:33 AM
 * Copyright: Branislav Maksin (c) 2022
 */

import React from 'react';
import { render } from '@testing-library/react';
import { expect } from '@jest/globals';
import {
  MemoryRouter,
  Routes,
  Route
} from 'react-router-dom';

import { MemberPage } from '../member';

// Mock Redux module
jest.mock('react-redux');

// Mock store instance
jest.mock('@store', () => ({}));

// Mock use data hook
jest.mock('../hooks/use-data/use-data');

describe('Member page component', () => {

  it('should render member page component', () => {

    // Render the component
    const { asFragment } = render(
      <MemoryRouter initialEntries={['/member/e837c9f5-247f-445f-bcc3-7d434348336b']}>
        <Routes>
          <Route
            element={<MemberPage />}
            path="/member/:memberId"
          />
        </Routes>
      </MemoryRouter>
    );

    // Match the snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});
