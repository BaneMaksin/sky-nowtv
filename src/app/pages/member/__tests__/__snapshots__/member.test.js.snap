// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member page component should render member page component 1`] = `
<DocumentFragment>
  <section>
    <h1
      class="title"
    >
      Member
    </h1>
    <header
      class="header"
    >
      <div
        class="context"
      >
        <h2
          class="title"
        >
          Martin Bradley
        </h2>
        <div
          class="text"
          data-testid="member-email"
        >
          <a
            href="mailto: mbradley0@google.it"
          >
            mbradley0@google.it
          </a>
        </div>
      </div>
      <img
        alt="Martin Bradley"
        class="image"
        data-testid="member-image"
        src="http://dummyimage.com/100x100.png/5fa2dd/ffffff"
      />
    </header>
    <h3
      class="title"
    >
      Messages
    </h3>
    <div
      class="wrapper"
    >
      <div
        class="card"
        data-testid="member-message"
      >
        <div
          class="message"
        >
          <p
            class="text"
          >
            Cras in purus eu magna vulputate luctus.
          </p>
        </div>
        <div
          class="author"
        >
          <small
            class="date"
          >
            26th Jan 2017 7:53
          </small>
        </div>
      </div>
      <div
        class="card"
        data-testid="member-message"
      >
        <div
          class="message"
        >
          <p
            class="text"
          >
            Morbi vestibulum, velit id pretium iaculis, diam erat fermentum justo, nec condimentum neque sapien placerat ante.
          </p>
        </div>
        <div
          class="author"
        >
          <small
            class="date"
          >
            25th Mar 2016 23:01
          </small>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
